vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-03-01/containerservice/models.go

```diff
@@ -2328,6 +2399,7 @@ func (mcp ManagedClusterProperties) MarshalJSON() ([]byte, error) {
 // ManagedClusterPropertiesAutoScalerProfile parameters to be applied to the cluster-autoscaler when
 // enabled
 type ManagedClusterPropertiesAutoScalerProfile struct {
+	BalanceSimilarNodeGroups      *bool   `json:"balance-similar-node-groups,omitempty"`
```

Comment:
AFAIK is this not the latest version, this should be a *string due to the latest 0.42 release, fixed here: https://github.com/Azure/azure-rest-api-specs/pull/8919#issuecomment-621453037
```suggestion
	BalanceSimilarNodeGroups      *string   `json:"balance-similar-node-groups,omitempty"`
```

Reply:
Hi @aristosvo 
I don't think @tombuildsstuff or @katbyte would find ok to update the SDK version in this PR.
If they are, happy to do so.

Reply:
+1, we will never update something in the vendor folder. i will be updating the go sdk version to 42 later today/tomorrow however

Reply:
@katbyte With regard to bumping the AKS API version, is that also a separate PR?